pennylane/optimize/adaptive.py

```diff
@@ -197,7 +197,7 @@ def step_and_cost(self, circuit, operator_pool, drain_pool=False, params_zero=Tr
                 operator_pool.remove(gate)
 
         params = np.array([gate.parameters[0] for gate in operator_pool], requires_grad=True)
-        qnode = qml.QNode(self._circuit, device)
+        qnode.func = self._circuit
         grads = qml.grad(qnode)(params, gates=operator_pool, initial_circuit=circuit.func)
 
         selected_gates = [operator_pool[np.argmax(abs(grads))]]
```

Comment:
Not related to this PR, but curious about why do we hardcode the optimizer method?

Reply:
No specific reason except to prevent potential confusion on defining an optimizer for another optimizer. We can support other optimizers and have a default one. Do you think it is important/needed?